src/Components/Steps/SignUp/SignUp.tsx

```diff
@@ -221,9 +223,9 @@ function SignUp() {
       await updateUser(newFormData);
       newFormData.signUpInfo.hasUser = true;
       setFormData(newFormData);
-      nextStep();
-    } catch {
+    } catch (e) {
       setHasServerError(true);
+      throw e;
```

Comment:
I did test locally that if an error is thrown, the submit successful state is not triggered.